File: Zad 3 python/Średnia ocen.py
# Pobranie liczby ocen
while True:
    try:
        liczba_ocen = int(input("Podaj liczbę ocen: "))

        if liczba_ocen > 0:
            break
        else:
            print("Liczba ocen musi być większa od 0!")

    except ValueError:
        print("Podaj poprawną liczbę!")

suma = 0

# Wprowadzanie ocen
for i in range(1, liczba_ocen + 1):

    while True:
        try:
            ocena = float(input(f"Podaj ocenę nr {i} (1-6): "))

            if 1 <= ocena <= 6:
                break
            else:
                print("Ocena musi być w zakresie 1-6!")

        except ValueError:
            print("Podaj poprawną ocenę!")

    suma += ocena

# Obliczanie średniej
srednia = suma / liczba_ocen

print("Średnia ocen wynosi:", srednia)

# Sprawdzenie zaliczenia
if srednia >= 3.0:
    print("Uczeń zaliczył przedmiot.")
else:
    print("Uczeń nie zaliczył przedmiotu.")

File: Zad3/Średnia ocen.cs
using System;

class Program
{
    static void Main()
    {
        // Pobranie liczby ocen
        Console.Write("Podaj liczbę ocen: ");

        int liczbaOcen;

        while (!int.TryParse(Console.ReadLine(), out liczbaOcen) || liczbaOcen <= 0)
        {
            Console.Write("Błędna wartość! Podaj dodatnią liczbę ocen: ");
        }

        double suma = 0;

        // Wprowadzanie ocen
        for (int i = 1; i <= liczbaOcen; i++)
        {
            Console.Write("Podaj ocenę nr " + i + " (1-6): ");

            double ocena;

            while (!double.TryParse(Console.ReadLine(), out ocena) || ocena < 1 || ocena > 6)
            {
                Console.Write("Błędna ocena! Podaj ocenę od 1 do 6: ");
            }

            suma += ocena;
        }

        // Obliczanie średniej
        double srednia = suma / liczbaOcen;

        // Wyświetlenie średniej
        Console.WriteLine("Średnia ocen wynosi: " + srednia);

        // Sprawdzenie zaliczenia
        if (srednia >= 3.0)
        {
            Console.WriteLine("Uczeń zaliczył przedmiot.");
        }
        else
        {
            Console.WriteLine("Uczeń nie zaliczył przedmiotu.");
        }
    }
}